itests/eth_filter_test.go

```diff
@@ -522,544 +483,193 @@ func TestEthGetLogs(t *testing.T) {
-		},
-		{
-			name: "find all events with topic2 of 46 or 50",
-			spec: newEthFilterBuilder().FromBlockEpoch(0).Topic2OneOf(paddedEthHash(paddedUint64(46)), paddedEthHash(paddedUint64(50))).Filter(),
+	testResponses := map[string]chan ethtypes.EthSubscriptionResponse{}
 
-			expected: []ExpectedEthLog{
-				{
-					Address: contract1,
-					Topics: []ethtypes.EthBytes{
-						EventMatrixContract.Ev["EventOneIndexedWithData"],
-						paddedUint64(46),
-					},
-					Data: paddedUint64(12),
-				},
-				{
-					Address: contract1,
-					Topics: []ethtypes.EthBytes{
-						EventMatrixContract.Ev["EventTwoIndexedWithData"],
-						paddedUint64(46),
-						paddedUint64(27),
-					},
-					Data: paddedUint64(19),
-				},
-				{
-					Address: contract1,
-					Topics: []ethtypes.EthBytes{
-						EventMatrixContract.Ev["EventTwoIndexedWithData"],
-						paddedUint64(46),
-						paddedUint64(14),
-					},
-					Data: paddedUint64(19),
-				},
-				{
-					Address: contract2,
-					Topics: []ethtypes.EthBytes{
-						EventMatrixContract.Ev["EventOneIndexedWithData"],
-						paddedUint64(50),
-					},
-					Data: paddedUint64(9),
-				},
-			},
-		},
+	// quit is used to signal that we're ready to start testing collected results
+	quit := make(chan struct{})
 
-		{
-			name: "find all events with topic1 of EventTwoIndexedWithData and topic3 of 27",
-			spec: newEthFilterBuilder().
-				FromBlockEpoch(0).
-				Topic1OneOf(paddedEthHash(EventMatrixContract.Ev["EventTwoIndexedWithData"])).
-				Topic3OneOf(paddedEthHash(paddedUint64(27))).
-				Filter(),
+	// Create all the filters
+	for _, tc := range testCases {
 
-			expected: []ExpectedEthLog{
-				{
-					Address: contract1,
-					Topics: []ethtypes.EthBytes{
-						EventMatrixContract.Ev["EventTwoIndexedWithData"],
-						paddedUint64(44),
-						paddedUint64(27),
-					},
-					Data: paddedUint64(19),
-				},
-				{
-					Address: contract1,
-					Topics: []ethtypes.EthBytes{
-						EventMatrixContract.Ev["EventTwoIndexedWithData"],
-						paddedUint64(46),
-						paddedUint64(27),
-					},
-					Data: paddedUint64(19),
-				},
-			},
-		},
+		// subscribe to topics in filter
+		subCh, err := client.EthSubscribe(ctx, "logs", &ethtypes.EthSubscriptionParams{Topics: tc.spec.Topics})
+		require.NoError(err)
+
+		responseCh := make(chan ethtypes.EthSubscriptionResponse, len(invocations))
+		testResponses[tc.name] = responseCh
+
+		// start a goroutine to forward responses from subscription to a buffered channel with guaranteed capacity
+		go func(subCh <-chan ethtypes.EthSubscriptionResponse, responseCh chan<- ethtypes.EthSubscriptionResponse, quit chan struct{}) {
+			defer func() {
+				close(responseCh)
+			}()
+			for {
+				select {
+				case resp := <-subCh:
+					responseCh <- resp
+				case <-quit:
+					return
+				case <-ctx.Done():
+					return
+				}
+			}
+		}(subCh, responseCh, quit)
+
+	}
+
+	// Perform all the invocations
+	messages := invokeAndWaitUntilAllOnChain(t, client, invocations)
```

Comment:
Good point. The ethereum event apis can use this flag, but these single miner tests aren't exercising it since. We could add futher tests that cause reversiions and ensure they are emitted in the etheraum event log apis.